src/main/java/lotto/domain/Rank.java

```diff
@@ -0,0 +1,49 @@
+package lotto.domain;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public enum Rank {
+    FIRST(6, false, 2_000_000_000),
+    SECOND(5, true, 30_000_000),
+    THIRD(5, false, 1_500_000),
+    FOURTH(4, false, 50_000),
+    FIFTH(3, false, 5_000),
+    NONE(0, false, 0);
+
+    private final int matchCount;
+    private final boolean matchBonus;
+    private final long price;
+
+    Rank(int matchCount, boolean matchBonus, long price) {
+        this.matchCount = matchCount;
+        this.matchBonus = matchBonus;
+        this.price = price;
+    }
+
+    public static Rank of(int matchCount, boolean matchBonus) {
+        Rank found = Arrays.stream(values())
+                .filter(rank -> rank.matchCount == matchCount)
+                .findFirst()
+                .orElse(NONE);
+
+        if (found == SECOND && THIRD.matchBonus == matchBonus) {
+            return THIRD;
+        }
+        return found;
+    }
+
+    public static long calculateTotalPrice(Map<Rank, Long> result) {
+        return result.entrySet().stream()
+                .mapToLong(set -> set.getKey().price * set.getValue())
```

Comment:
[이전 코멘트](https://github.com/woowacourse/java-lotto/pull/542#discussion_r1954612497)에 대한 의도는 아래와 같이 변경하는 거였어요! 

Rank 객체(= set.getKey())에게 메시지를보내보자는 의도였어요! 😃
예를 들자면 아래와 같아요! 
```suggestion
                .mapToLong(set -> set.getKey().multiplyPrice(set.getValue()))
```

calculateTotalPrice로 정의해주신 내용은 기존 위치로 원복시켜주시면 좋을 것 같아요. 😃